### PR TITLE
🔫  891 Integrate GRPC triggering method

### PR DESCRIPTION
**Pull request by xoscar**
This PR enables the GRPC triggering method to be used.
It includes a new way of handling step forms, allowing us to use the same form for creation and editing.

## Changes

- New triggering method.
- Creation and editing implementation.

## Fixes

- https://github.com/kubeshop/tracetest/issues/891

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

https://www.loom.com/share/3333b9c958bc4894bf5c2c4112087f04

### CODE REVIEW

File: web/src/components/CreateTestPlugins/Default/steps/BasicDetails/BasicDetailsDemoHelper.tsx [hunk 1]
```diff
@@ -1,35 +1,24 @@
...
   const handleOnDemoClick = ({key}: {key: string}) => {
     CreateTestAnalyticsService.onDemoTestClick();
     const demo = DemoTestExampleList.find(({name}) => name === key)!;
     onSelectDemo(demo);
-
-    const {description, name} = demo;
-
-    form.setFieldsValue({
-      name,
-      description,
```
Comment (xoscar) [inline web/src/components/CreateTestPlugins/Default/steps/BasicDetails/BasicDetailsDemoHelper.tsx:28]: Moved this to an upper level as well

File: web/src/components/CreateTestPlugins/Rpc/steps/RequestDetails/RequestDetails.tsx [hunk 2]
```diff
@@ -1,65 +1,56 @@
...
 
   const handleNext = useCallback(() => {
     form.submit();
   }, [form]);
 
   const handleSubmit = useCallback(
-    (values: IRequestDetailsValues) => {
-      console.log('@@values', values);
-      onNext();
+    async ({protoFile, message, metadata, method, auth, url}: IRpcValues) => {
```
Comment (xoscar) [inline web/src/components/CreateTestPlugins/Rpc/steps/RequestDetails/RequestDetails.tsx:23]: wiring up the rpc flow

File: web/src/components/EditTestForm/EditRequestDetails/Grpc/Grpc.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import {useCallback, useEffect} from 'react';
```
Comment (xoscar) [inline web/src/components/EditTestForm/EditRequestDetails/Grpc/Grpc.tsx:1]: Utilizes the same form component from the creation flow

File: web/src/services/Test.service.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  return Validator.required(name) && Validator.required(description);
+};
+
+const TriggerServiceMap = {
+  [TriggerTypes.grpc]: RpcService,
+  [TriggerTypes.http]: HttpService,
+} as const;
+
+const TestService = () => ({
+  async getRequest(type: TriggerTypes, draft: TDraftTest, original?: TTest): Promise<TRawTest> {
```
Comment (xoscar) [inline web/src/services/Test.service.ts:36]: Both validates and retrieves the request

File: web/src/services/Test.service.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import {
+  TRawTest,
+  TTest,
+  TDraftTest,
+  // IBasicValues,
```
Comment (jorgeepc) [inline web/src/services/Test.service.ts:5]: Do we want to keep this?

**Review by jorgeepc — COMMENTED** [review]
Love what you did here @xoscar 🔥 Added a couple of comments.

**Review by jorgeepc — APPROVED** [review]

**Pull request merged**